src/cogent3/util/deserialise.py

```diff
@@ -262,17 +262,28 @@ def deserialise_likelihood_function(data):
     """returns a cogent3 likelihood function instance"""
     data.pop("version", None)
     model = deserialise_substitution_model(data.pop("model"))
-    aln = deserialise_seq_collections(data.pop("alignment"))
     tree = deserialise_tree(data.pop("tree"))
     constructor_args = data.pop("likelihood_construction")
     motif_probs = data.pop("motif_probs")
     param_rules = data.pop("param_rules")
     name = data.pop("name", None)
     lf = model.make_likelihood_function(tree, **constructor_args)
     lf.set_name(name)
-    lf.set_alignment(aln)
+    lf = model.make_likelihood_function(tree, **constructor_args)
+    if isinstance(constructor_args["loci"], list):
+        align = data["alignment"]
+        loci = [deserialise_seq_collections(align[k]) for k in align]
+        lf.set_alignment(loci)
+        mprobs = [motif_probs[k] for k in motif_probs]
+        with lf.updates_postponed():
+            for motif_probs in mprobs:
+                lf.set_motif_probs(motif_probs)
+    else:
+        aln = deserialise_seq_collections(data.pop("alignment"))
+        lf.set_alignment(aln)
```

Comment:
move this `lf.set_alignment(aln)` to after the if/else block, but before the `with` clause from above and add
```python
mprobs = [motif_probs]
```
inside the `else` block.